### PR TITLE
Initialize interactions on device

### PR DESCRIPTION
**Pull request by amandalund**
This makes sure the `Interaction`s are initialized at the start of the simulation, as some members (`secondaries`) might otherwise be used uninitialized. This could possibly be the cause of the demo-loop error seen in #374.

### CODE REVIEW

**Comment by sethrj** [comment]
@amandalund This fixes the crashes on AMD! 🤩  Nice work!!

File: src/sim/TrackData.hh [hunk 1]
```diff
@@ -169,7 +169,13 @@ resize(StateData<Ownership::value, M>*                               data,
...
     resize(&data->step_length, size);
     resize(&data->energy_deposition, size);
-    resize(&data->interactions, size);
+
+    // Initialize empty interactions
+    StateCollection<Interaction, Ownership::value, MemSpace::host> interactions;
+    std::vector<Interaction> initial_state(size, Interaction{});
+    make_builder(&interactions)
+        .insert_back(initial_state.begin(), initial_state.end());
+    data->interactions = interactions;
```
Comment (sethrj) [inline src/sim/TrackData.hh:178]: It would probably be better to instantiate the `Filler` and do the initializations on device rather than copy, or to call a small kernel to clear all the track data as part of `resize`. We can do that in another MR though, I'm too excited about AMD working 😂

**Review by sethrj — APPROVED** [review]

File: src/sim/TrackData.hh (hunk 1 above)
Comment (amandalund) [reply src/sim/TrackData.hh:178]: Haha, sounds good! :)
